src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,362 @@
+            }
+
+            string remoteAddressString = remoteAddressAndPort.Substring(0, indexOfColon);
+            remoteIPAddress = ParseHexIPAddress(remoteAddressString);
+
+            string portString = remoteAddressAndPort.Substring(indexOfColon + 1, remoteAddressAndPort.Length - (indexOfColon + 1));
+            if (!int.TryParse(portString, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out remotePort))
+            {
+                throw new InvalidOperationException("Couldn't parse remote port number " + portString);
+            }
+        }
+
+        private static void ParseAddressAndPort(string colonSeparatedAddress, out IPAddress ipAddress, out int port)
+        {
+            int indexOfColon = colonSeparatedAddress.IndexOf(':');
+            if (indexOfColon == -1)
+            {
+                throw new InvalidOperationException("Parsing error. No colon in " + colonSeparatedAddress);
+            }
+
+            string remoteAddressString = colonSeparatedAddress.Substring(0, indexOfColon);
+            ipAddress = ParseHexIPAddress(remoteAddressString);
+
+            string portString = colonSeparatedAddress.Substring(indexOfColon + 1, colonSeparatedAddress.Length - (indexOfColon + 1));
+            if (!int.TryParse(portString, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out port))
+            {
+                throw new InvalidOperationException("Couldn't parse remote port number " + portString);
+            }
+        }
+
+        private static IPAddress ParseHexIPAddress(string remoteAddressString)
+        {
+            if (remoteAddressString.Length <= 8) // IPv4 Address
+            {
+                return ParseIPv4HexString(remoteAddressString);
+            }
+            else // IPv6 Address
+            {
+                return ParseIPv6HexString(remoteAddressString);
+            }
+        }
+
+        // Simply converst the hex string into a long and uses the IPAddress(long) constructor.
+        // Strings passed to this method must be 8 or less characters in length (32-bit address).
+        private static IPAddress ParseIPv4HexString(string hexAddress)
+        {
+            IPAddress ipAddress;
+            long addressValue;
+            if (!long.TryParse(hexAddress, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out addressValue))
+            {
+                throw new InvalidOperationException("Couldn't parse long from address string " + hexAddress);
+            }
+            ipAddress = new IPAddress(addressValue);
+            return ipAddress;
+        }
+
+        // Parses a 128-bit IPv6 Address stored as a 32-character hex number.
+        // Strings passed to this must be 32 characters in length.
+        private static IPAddress ParseIPv6HexString(string hexAddress)
+        {
+            IPAddress ipAddress;
+            // Dangerous Parsing Code
```

Comment:
And expensive, creating a string for each hex digit.  It'd be better to do something like https://github.com/dotnet/corefx/blob/2ea446cc13790ae5ccdff1b2df155d2d6680e6d9/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs#L38 but without all of the whitespace goo stuff.